.circleci/config.yml

```diff
@@ -252,17 +252,26 @@ commands:
             echo "Using SSH Config Dir '$SSH_CONFIG_DIR'"
             mkdir -p $SSH_CONFIG_DIR
             echo -e 'github.com ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEmKSENjQEezOmxkZMy7opKgwFB9nkt5YRrYMjNuG5N87uRgg6CLrbo5wAdT/y6v0mKV0U2w0WZ2YB/++Tpockg=\n' >> "$SSH_CONFIG_DIR/known_hosts"
+            chmod 0600 $SSH_CONFIG_DIR/known_hosts
+
             mkdir -p $CIRCLE_WORKING_DIRECTORY
-            git clone --depth 1 "$CIRCLE_REPOSITORY_URL" --branch "$CIRCLE_BRANCH" $CIRCLE_WORKING_DIRECTORY
+            git clone --depth 1 "$CIRCLE_REPOSITORY_URL" $CIRCLE_WORKING_DIRECTORY
             cd $CIRCLE_WORKING_DIRECTORY
 
-      - run:
-          name: Fetch tag from origin if present
-          command: |
-            if [ -n "$CIRCLE_TAG" ] ; then
-              git fetch --force origin "refs/tags/${CIRCLE_TAG}:refs/tags/${CIRCLE_TAG}"
+            if [ -n "$CIRCLE_TAG" ]; then
+              echo 'Fetch tag'
+              git fetch --depth 5 --force origin "+refs/tags/${CIRCLE_TAG}:refs/tags/${CIRCLE_TAG}"
+            elif [[ $(echo $CIRCLE_BRANCH | grep -E ^pull\/[0-9]+$) ]]; then
```

Comment:
Does this cause any issue if we have branch with same name in the actual repo?

Reply:
yeah, it will cause problem. I will fix that.